error: expected `,`
 --> $DIR/args.rs:7:21
  |
7 | #[auto_enum(Iterator;)] //~ ERROR expected `,`
  |                     ^

error: expected identifier
  --> $DIR/args.rs:15:22
   |
15 | #[auto_enum(Iterator,;)] //~ ERROR expected identifier
   |                      ^

error: expected `,`
  --> $DIR/args.rs:26:23
   |
26 |     #[auto_enum(marker{f}, Iterator)] //~ ERROR expected `,`
   |                       ^^^

error: expected `,`
  --> $DIR/args.rs:34:23
   |
34 |     #[auto_enum(marker[f], Iterator)] //~ ERROR expected `,`
   |                       ^^^

error: expected `,`
  --> $DIR/args.rs:42:23
   |
42 |     #[auto_enum(marker(f), Iterator)] //~ ERROR expected `,`
   |                       ^^^

error: duplicate `marker` argument
  --> $DIR/args.rs:50:38
   |
50 |     #[auto_enum(marker = f, marker = g, Iterator)] //~ ERROR duplicate `marker` argument
   |                                      ^

error: expected identifier
  --> $DIR/args.rs:58:25
   |
58 |     #[auto_enum(marker =, Iterator)] //~ ERROR expected identifier
   |                         ^

error: expected `,`
  --> $DIR/args.rs:66:28
   |
66 |     #[auto_enum(marker = f t, Iterator)] //~ ERROR expected `,`
   |                            ^
