error: expected `,`
 --> $DIR/args.rs:8:21
  |
8 | #[auto_enum(Iterator;)] //~ ERROR expected `,`
  |                     ^

error: expected identifier
  --> $DIR/args.rs:16:22
   |
16 | #[auto_enum(Iterator,;)] //~ ERROR expected identifier
   |                      ^

error: expected `,`
  --> $DIR/args.rs:27:23
   |
27 |     #[auto_enum(marker{f}, Iterator)] //~ ERROR expected `,`
   |                       ^^^

error: expected `,`
  --> $DIR/args.rs:35:23
   |
35 |     #[auto_enum(marker[f], Iterator)] //~ ERROR expected `,`
   |                       ^^^

error: expected `,`
  --> $DIR/args.rs:43:23
   |
43 |     #[auto_enum(marker(f), Iterator)] //~ ERROR expected `,`
   |                       ^^^

error: duplicate `marker` argument
  --> $DIR/args.rs:51:38
   |
51 |     #[auto_enum(marker = f, marker = g, Iterator)] //~ ERROR duplicate `marker` argument
   |                                      ^

error: expected identifier
  --> $DIR/args.rs:59:25
   |
59 |     #[auto_enum(marker =, Iterator)] //~ ERROR expected identifier
   |                         ^

error: expected `,`
  --> $DIR/args.rs:67:28
   |
67 |     #[auto_enum(marker = f t, Iterator)] //~ ERROR expected `,`
   |                            ^

error: aborting due to 8 previous errors

