error: expected one of `,`, or `::`, found `;`
 --> $DIR/args.rs:8:21
  |
8 | #[auto_enum(Iterator;)] //~ ERROR expected one of `,`, or `::`, found `;`
  |                     ^

error: expected one of `,`, `::`, or identifier, found `;`
  --> $DIR/args.rs:16:22
   |
16 | #[auto_enum(Iterator,;)] //~ ERROR expected one of `,`, `::`, or identifier, found `;`
   |                      ^

error: invalid delimiter
  --> $DIR/args.rs:27:23
   |
27 |     #[auto_enum(marker{f}, Iterator)] //~ ERROR invalid delimiter
   |                       ^^^

error: invalid delimiter
  --> $DIR/args.rs:35:23
   |
35 |     #[auto_enum(marker[f], Iterator)] //~ ERROR invalid delimiter
   |                       ^^^

error: multiple `marker` option
  --> $DIR/args.rs:43:35
   |
43 |     #[auto_enum(marker(f), marker(g), Iterator)] //~ ERROR multiple `marker` option
   |                                   ^

error: empty `marker` option
  --> $DIR/args.rs:51:23
   |
51 |     #[auto_enum(marker(), Iterator)] //~ ERROR empty `marker` option
   |                       ^^

error: multiple identifier in `marker` option
  --> $DIR/args.rs:59:27
   |
59 |     #[auto_enum(marker(f, g), Iterator)] //~ ERROR multiple identifier in `marker` option
   |                           ^

error: multiple identifier in `marker` option
  --> $DIR/args.rs:67:26
   |
67 |     #[auto_enum(marker(f t), Iterator)] //~ ERROR multiple identifier in `marker` option
   |                          ^

error: multiple `marker` option
  --> $DIR/args.rs:75:34
   |
75 |     #[auto_enum(marker=f, marker=g, Iterator)] //~ ERROR multiple `marker` option
   |                                  ^

error: expected an identifier, found `,`
  --> $DIR/args.rs:83:24
   |
83 |     #[auto_enum(marker=, Iterator)] //~ ERROR empty `marker` option
   |                        ^

error: expected `,`, found `t`
  --> $DIR/args.rs:91:26
   |
91 |     #[auto_enum(marker=f t, Iterator)] //~ ERROR expected `,`, found `t`
   |                          ^

error: aborting due to 11 previous errors

