error: unexpected token
  --> $DIR/attribute.rs:11:16
   |
11 |         #[never(foo)] //~ ERROR unexpected token
   |                ^

error: unexpected token
  --> $DIR/attribute.rs:21:17
   |
21 |         #[nested(foo)] //~ ERROR unexpected token
   |                 ^

error: #[rec] has been removed and replaced with #[nested]
  --> $DIR/attribute.rs:31:9
   |
31 |         #[rec] //~ ERROR #[rec] has been removed and replaced with #[nested]
   |         ^^^^^^
