error: expected one of `)`, `,`, `.`, `?`, or an operator, found `..`
  --> $DIR/compile-fail.rs:10:26
   |
10 |         0 => marker!(1..8..), //~ ERROR expected one of `)`, `,`, `.`, `?`, or an operator, found `..`
   |                          ^^ expected one of `)`, `,`, `.`, `?`, or an operator here

error: `if` expression missing an else clause
  --> $DIR/compile-fail.rs:19:12
   |
19 |     } else if x > 3 {
   |            ^^

error: `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in this statement
  --> $DIR/compile-fail.rs:28:5
   |
28 | /     let iter = match x {
29 | |         //~^ ERROR `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in this statement
30 | |         _ if y < 0 => return y..=0,
31 | |         _ => 2..=10,
32 | |     };
   | |______^

error: `#[auto_enum]` is required two or more branches or marker macros in total, there is no branch or marker macro in this statement
  --> $DIR/compile-fail.rs:43:5
   |
43 | /     let iter = match x {
44 | |         //~^ ERROR `#[auto_enum]` is required two or more branches or marker macros in total, there is no branch or marker macro in this statement
45 | |         _ if y < 0 => return y..=0,
46 | |         _ => return 2..=10,
47 | |     };
   | |______^

error[E0061]: this function takes 1 parameter but 2 parameters were supplied
 --> $DIR/compile-fail.rs:7:1
  |
7 | #[auto_enum(Iterator)]
  | ^^^^^^^^^^^^^^^^^^^^^^
  | |
  | defined here
  | expected 1 parameter

For more information about this error, try `rustc --explain E0061`.
