error: expected one of `)`, `,`, `.`, `?`, or an operator, found `..`
 --> $DIR/compile-fail.rs:8:26
  |
8 |         0 => marker!(1..8..), //~ ERROR expected one of `)`, `,`, `.`, `?`, or an operator, found `..`
  |                          ^^ expected one of `)`, `,`, `.`, `?`, or an operator here

error: `if` expression missing an else clause
  --> $DIR/compile-fail.rs:17:12
   |
17 |     } else if x > 3 {
   |            ^^

error: `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in this statement
  --> $DIR/compile-fail.rs:26:5
   |
26 | /     let iter = match x {
27 | |         //~^ ERROR `#[auto_enum]` is required two or more branches or marker macros in total, there is only one branch or marker macro in this statement
28 | |         _ if y < 0 => return y..=0,
29 | |         _ => 2..=10,
30 | |     };
   | |______^

error: `#[auto_enum]` is required two or more branches or marker macros in total, there is no branch or marker macro in this statement
  --> $DIR/compile-fail.rs:41:5
   |
41 | /     let iter = match x {
42 | |         //~^ ERROR `#[auto_enum]` is required two or more branches or marker macros in total, there is no branch or marker macro in this statement
43 | |         _ if y < 0 => return y..=0,
44 | |         _ => return 2..=10,
45 | |     };
   | |______^

error[E0061]: this function takes 1 parameter but 2 parameters were supplied
 --> $DIR/compile-fail.rs:5:1
  |
5 | #[auto_enum(Iterator)]
  | ^^^^^^^^^^^^^^^^^^^^^^
  | |
  | defined here
  | expected 1 parameter
