error: expected identifier
 --> $DIR/args.rs:7:22
  |
7 | #[enum_derive(Clone, =>)] //~ ERROR expected identifier
  |                      ^^

error: expected identifier
  --> $DIR/args.rs:13:25
   |
13 | #[enum_derive(foo::bar::!)] //~ ERROR expected identifier
   |                         ^

error: expected `,`
  --> $DIR/args.rs:19:25
   |
19 | #[enum_derive(Clone, Foo:)] //~ ERROR expected `,`
   |                         ^

error: expected `,`
  --> $DIR/args.rs:25:21
   |
25 | #[enum_derive(Clone Foo)] //~ ERROR expected `,`
   |                     ^^^
