error: expected identifier
 --> $DIR/args.rs:5:22
  |
5 | #[enum_derive(Clone, =>)] //~ ERROR expected identifier
  |                      ^^

error: expected identifier
  --> $DIR/args.rs:11:25
   |
11 | #[enum_derive(foo::bar::!)] //~ ERROR expected identifier
   |                         ^

error: expected `,`
  --> $DIR/args.rs:17:25
   |
17 | #[enum_derive(Clone, Foo:)] //~ ERROR expected `,`
   |                         ^

error: expected `,`
  --> $DIR/args.rs:23:21
   |
23 | #[enum_derive(Clone Foo)] //~ ERROR expected `,`
   |                     ^^^
