error: an enum with units variant is not supported
 --> $DIR/variant.rs:9:5
  |
9 |     A, //~ ERROR an enum with units variant is not supported
  |     ^

error: an enum with discriminants is not supported
  --> $DIR/variant.rs:15:9
   |
15 |     A = 2, //~ ERROR an enum with discriminants is not supported
   |         ^

error: an enum with named fields variant is not supported
  --> $DIR/variant.rs:21:5
   |
21 |     A { x: A }, //~ ERROR an enum with named fields variant is not supported
   |     ^^^^^^^^^^

error: a variant with zero fields is not supported
  --> $DIR/variant.rs:27:6
   |
27 |     A(), //~ ERROR a variant with zero fields is not supported
   |      ^^

error: a variant with multiple fields is not supported
  --> $DIR/variant.rs:34:6
   |
34 |     B(A, B), //~ ERROR a variant with multiple fields is not supported
   |      ^^^^^^

error: cannot be implemented for enums with no variants
  --> $DIR/variant.rs:38:1
   |
38 | enum Enum7 {} //~ ERROR cannot be implemented for enums with less than two variants
   | ^^^^^^^^^^^^^
